internal/services/send_receiver_wallets_invite_service.go

```diff
@@ -145,24 +157,79 @@ func (s SendReceiverWalletInviteService) SendInvite(ctx context.Context) error {
 		}
 
 		msgsToInsert = append(msgsToInsert, msgToInsert)
+
+		// We don't want to update the `invitation_sent_at` for receiver wallets that we've sent the invitation SMS
+		// because there's no way to calculate how many times we've resent the invitation SMS since
+		// the first invitation if we update it.
+		if rwa.ReceiverWallet.InvitationSentAt == nil && msgToInsert.Status == data.SuccessMessageStatus {
+			receiverWalletIDs = append(receiverWalletIDs, rwa.ReceiverWallet.ID)
+		}
+	}
+
+	return db.RunInTransaction(ctx, s.models.DBConnectionPool, nil, func(dbTx db.DBTransaction) error {
+		if _, err := s.models.ReceiverWallet.UpdateInvitationSentAt(ctx, dbTx, receiverWalletIDs...); err != nil {
+			return fmt.Errorf("updating receiver wallets' invitation sent at: %w", err)
+		}
+
+		if err := s.models.Message.BulkInsert(ctx, dbTx, msgsToInsert); err != nil {
+			return fmt.Errorf("error inserting messages in the database: %w", err)
+		}
+
+		return nil
+	})
+}
+
+func (s SendReceiverWalletInviteService) shouldSendInvitationSMS(ctx context.Context, organization *data.Organization, rwa *data.ReceiverWalletAsset) bool {
+	// We've never sent a Invitation SMS
+	if rwa.ReceiverWallet.InvitationSentAt == nil {
+		return true
 	}
 
-	if err := s.models.Message.BulkInsert(ctx, msgsToInsert); err != nil {
-		return fmt.Errorf("error inserting messages in the database: %w", err)
+	// If organization's SMS Resend Interval is nil and we've sent the invitation message to the receiver, we won't resend it.
+	if organization.SMSResendInterval == nil && rwa.ReceiverWallet.InvitationSentAt != nil {
+		return false
```

Comment:
Do we want to log the reason for not sending the SMS here ?

Reply:
It would be great to log it! Thank you.